### ISSUE TITLE
Facebook share

### ISSUE THREAD

**Issue by nolanRatu** [jmacmullin/BCDShareSheet#1]
Hi thanks for the BCDShareSheet library, it's awesome. I have it working just fine. But, how do i set it up to post a message on facebook? maybe on the users page. I have created the facebook app, but when i select facebook to share it loads and closes and starts my ios app again, thats what happens when i am already logged in. when im not i go through the login process and then it takes me back to my app.

I want the facebook share to take the user to their page and then have infomation about my app like the email share, so they can post it on their page. Any ideas?

**Comment by jmacmullin** [comment]
It sounds like you might not have implemented the application:openURL:sourceApplication: method in your App's delegate. In your app's delegate you'll need an implementation of this method that passes the Facebook callback URL to BCDShareSheet.

Have a look at the sample app for details - but you should have something like this:

``` objective-c
- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation {
    if ([[url scheme] hasPrefix:@"fb"]) {
        return [[BCDShareSheet sharedSharer] openURL:url];
    } else {
        return NO; // handle any other URLs your app responds to here.
    }
}
```

**Comment by nolanRatu** [comment]
Yep i have that in my delegate. But when i select the facebook share it opens Facebook then it takes me to my app again. is the facebook share suppose to take me to my own facebook share page?

**Comment by jmacmullin** [comment]
When you attempt to share something via Facebook, you should be redirected to Facebook (either the app or the website) in order to login. Once you've been authenticated you should be redirected back to your app and your app delegate's application:openURL:sourceApplication: method should be called with a fb:// URL. I'd suggest setting a breakpoint in this method and seeing what is happening when you're returned to your app. Is the method getting called? What is the value of the 'url' parameter?

**Comment by nolanRatu** [comment]
I have followed your tutorial, so when i select facebook to share,it does not bring up a window like the email or twitter for me to share. this is what i want to happen.

**Comment by nolanRatu** [comment]
This is what i want to show when someone clicks on the facebook share:
NSMutableDictionary\* params = [NSMutableDictionary dictionaryWithObjectsAndKeys:
  kAppId, @"app_id",
  @"https://developers.facebook.com/docs/reference/dialogs/", @"link",
  @"http://fbrell.com/f8.jpg", @"picture",
  @"Facebook Dialogs", @"name",
  @"Reference Documentation", @"caption",
  @"Using Dialogs to interact with users.", @"description",
  @"Facebook Dialogs are so easy!",  @"message",
  nil];

[_facebook dialog:@"feed" andParams:params andDelegate:self];
